content_sync/utils.py

```diff
@@ -79,10 +88,156 @@ def check_mandatory_settings(mandatory_settings):
 
 def move_s3_object(from_path, to_path):
     """Move an S3 object from one path to another"""
-    s3 = boto3.resource("s3")
+    s3 = get_boto3_resource("s3")
     bucket = settings.AWS_STORAGE_BUCKET_NAME
     extra_args = {"ACL": "public-read"}
     s3.meta.client.copy(
         {"Bucket": bucket, "Key": from_path}, bucket, to_path, extra_args
     )
     s3.Object(bucket, from_path).delete()
+
+
+def get_template_vars():
+    """Get an object with all the template vars we need in pipelines based on env"""
+    base_vars = {
+        "preview_bucket_name": settings.AWS_PREVIEW_BUCKET_NAME,
+        "publish_bucket_name": settings.AWS_PUBLISH_BUCKET_NAME,
+        "storage_bucket_name": settings.AWS_STORAGE_BUCKET_NAME,
+        "artifacts_bucket_name": "ol-eng-artifacts",
+    }
+    default_vars = {
+        "resource_base_url_draft": "",
+        "resource_base_url_live": "",
+        "ocw_studio_url": settings.SITE_BASE_URL,
+    }
+    default_vars.update(base_vars)
+    dev_vars = {
+        "resource_base_url_draft": settings.RESOURCE_BASE_URL_DRAFT,
+        "resource_base_url_live": settings.RESOURCE_BASE_URL_LIVE,
+        "ocw_studio_url": "http://10.1.0.102:8043",
+    }
+    dev_vars.update(base_vars)
+    return dev_vars if is_dev() else default_vars
+
+
+def get_theme_branch():
+    """
+    Gets the branch to use of ocw-hugo-themes in pipelines, defaulting to settings.GITHUB_WEBHOOK_BRANCH
+    if settings.ENVIRONMENT is anything but "dev," otherwise take the value of settings.OCW_HUGO_THEMES_BRANCH
+    """
+    github_webhook_branch = settings.GITHUB_WEBHOOK_BRANCH
```

Comment:
You mean instead of `settings.GITHUB_WEBHOOK_BRANCH`?  Our Heroku environments currently only have `settings.GITHUB_WEBHOOK_BRANCH` which is set to `release-candidate` in RC and `release` in Production.  In these environments, we want the branch of `ocw-hugo-themes` and `ocw-hugo-projects` to match exactly.  When you're developing locally, you might want to try one branch of `ocw-hugo-themes` with another branch of `ocw-hugo-projects` where you're working on configuration changes to the `config.yaml` file.  This allows you to push up your changes to your development branch, run your pipeline and see the output as built by Concourse.  Most of these scenarios can just be tested by running the hugo / node dev servers in `ocw-hugo-themes`, but with this alternate theme rendering stuff I'm working on I need to be able to test loading a different theme with a different config file and I want to be able to test having Concourse output the site to a different path.